code/game/objects/effects/spawners/minor_variation.dm

```diff
@@ -0,0 +1,58 @@
+/*
+Makes variations possible with code only.
+This is mainly to prevent creating new map files when creating smaller variations that aren't worth a map file. - Hopek
+*/
+
+/obj/effect/variation
+	name = "variation"
+	icon = 'icons/effects/landmarks_variation.dmi'
+	icon_state = "generic_green"
+	var/variations // number of variations
+	var/choice // chosen variation by choice()
+	var/chosen // has a choice been made for this place?
+	var/location_place // location of placement calculated from reference
+	var/ref_x // x of reference
+	var/ref_y // y of reference
+	var/ref_z // z of reference
+
+/obj/effect/variation/Initialize()
+	. = ..()
+	chosen = FALSE
+
+/obj/effect/variation/proc/place(x, y , item , direction)
+	if(!x)
+		x = 0
+	if(!y)
+		y = 0
+	if(item)
+		location_place = locate((ref_x + x) , (ref_y + y) , ref_z)
```

Comment:
```suggestion
		var/turf/location_place = locate((ref_x + x) , (ref_y + y) , ref_z)
```

Reply:
What's the difference at least on the top its easier to find the var but the object gets deleted after the spawning happens.

Reply:
You should declare the var when you need it so it's easy to see that it's not being used for anything else. Doing it in the declaration might cause someone to think it's used for something else (Which is what I did initially)